### PR TITLE
TS-32412 Support proxy servers

### PR DESCRIPTION
**Pull request by AlEscher**
Addresses issue [TS-32412](https://cqse.atlassian.net/browse/TS-32412)

- [x] No findings in [Teamscale](https://demo.teamscale.com/activity.html#/teamscale-javascript-profiler/) (c.f. vote of the Teamscale Bot). Flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.
- [x] Changes are tested adequately
- [x] Profilers's README.md and the Teamscale documentation are updated in case of user-visible changes
- [x] CHANGELOG.md of the affected packages are updated
- [ ] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)

### CODE REVIEW

**Comment by AlEscher** [comment]
I was unsure on how exactly the versioning in the CHANGELOG.md works, please correct me if I am wrong.

File: packages/teamscale-coverage-collector/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@ We use [Semantic Versioning](https://semver.org/).
 
 # New Release
 
+# 0.0.1-beta.45
```
Comment (stahlbauer) [inline packages/teamscale-coverage-collector/CHANGELOG.md:5]: Remove this. We are already at .47. New entries are supposed to be under `# New Release`.

File: packages/teamscale-coverage-collector/src/upload/ProxyUpload.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import {ConfigParameters} from "../utils/ConfigParameters";
+import {AxiosProxyConfig} from "axios";
+import url from "url";
+
+/**
+ * Creates an AxiosProxyConfig object if proxy variables are provided.
+ */
+export function addProxyOptions(config: ConfigParameters): AxiosProxyConfig | undefined {
```
Comment (stahlbauer) [inline packages/teamscale-coverage-collector/src/upload/ProxyUpload.ts:8]: The name of this function is misleading. `extractProxyOptions`?

File: packages/teamscale-coverage-collector/src/utils/ConfigParameters.ts [hunk 3]
```diff
@@ -150,6 +152,10 @@ export function buildParameterParser(): ArgumentParser {
 		help: '(optional): The path within the storage location between the default path and the uploaded artifact.',
 		default: process.env.ARTIFACTORY_PATH_SUFFIX
 	});
+	parser.add_argument('--http-proxy', {
```
Comment (stahlbauer) [inline packages/teamscale-coverage-collector/src/utils/ConfigParameters.ts:155]: It is unclear if also HTTPS is supported.

**Review by stahlbauer — COMMENTED** [review]
Thanks for contributing :) I have added some notes.

**Review by stahlbauer — APPROVED** [review]
Thanks for contributing. Approved.

**Pull request merged**